1.basic/6.validation/solution/zomes/exercise/src/lib.rs

```diff
@@ -19,81 +19,85 @@ fn add_estimate(external_estimate: Estimate) -> ExternResult<HeaderHashB64> {
 fn validate(op: Op) -> ExternResult<ValidateCallbackResult> {
     match op {
         Op::StoreEntry {
-            header:
-                SignedHashed {
-                    hashed:
-                        HoloHashed {
-                            content: header, ..
-                        },
-                    ..
-                },
+            header: SignedHashed {
+                hashed: HoloHashed {
+                    content: header, ..
+                }, ..
+            },
             entry,
-        } => match header.app_entry_type() {
-            Some(app_entry_type) => {
-                let this_zome = zome_info()?;
-                let matches_entry_def =
-                    this_zome.matches_entry_def_id(app_entry_type, Estimate::entry_def_id());
-
-                if matches_entry_def {
-                    let estimate = Estimate::try_from(&entry)?;
-
-                    let value: u8 = estimate.value;
-                    if is_estimate_invalid(value) {
-                        return Ok(ValidateCallbackResult::Invalid(
-                            "No a correct value".to_string(),
-                        ));
-                    }
-                    return Ok(ValidateCallbackResult::Valid);
-                } else {
-                    Ok(ValidateCallbackResult::Invalid(format!(
-                        "Not a Estimate but a {:?}",
-                        header.entry_type()
-                    )))
-                }
+        } => validate_entry_creation(header, entry),
+        Op::StoreElement {
+            element: Element {
+                entry: element_entry,
+                ..
             }
-            None => Ok(ValidateCallbackResult::Valid),
-        },
-        _ => Ok(ValidateCallbackResult::Valid),
+        } => validate_element_entry(element_entry),
```

Comment:
If Elements are Header + Entry, then why is there no header to validate in this branch? (When I tried, I could not get a typed header like `EntryCreationHeader` that would allow me to validate anything useful.